### DISCUSSION TITLE
Subscribe to the tick event

### DISCUSSION THREAD

**Discussion by qualterz** [saul/demofile-net#70] [Q&A] [answered]
Is there a way to subscribe to the tick event?

**Comment by saul** [comment, marked as answer]
I assume you're using ReadAllAsync? In which case you can add a callback to `demo.OnProgress`. You can check if the tick has changed there - that's fired after each and every command in the demo is parsed.

**Reply by qualterz** [reply]
That's an acceptable answer, thank you!
